docs/pages/guides/server-components.mdx

```diff
@@ -0,0 +1,400 @@
+
+```tsx app/index.tsx
+export default function Index() {
+  return (
+    <Button
+      title="Press me"
+      onPress={async () => {
+        'use server';
+        // This code runs on the server.
+        console.log('Button pressed');
+        return '...';
+      }}
+    />
+  );
+}
+```
+
+We can create a client component to invoke our server action:
+
+```tsx components/button.tsx
+'use client';
+import { Text } from 'react-native';
+
+export default function Button({ title, onPress }) {
+  return <Text onPress={() => onPress()}>{title}</Text>;
+}
+```
+
+Server Actions can also be defined in a standalone file (with "use server" at the top) and imported from client components:
+
+```tsx components/server-actions.tsx
+'use server';
+
+export async function callAction() {
+  // ...
+}
+```
+
+These can be used in client components:
+
+```tsx components/button.tsx
+import { Text } from 'react-native';
+import { callAction } from './server-actions';
+
+export default function Button({ title }) {
+  return <Text onPress={() => callAction()}>{title}</Text>;
+}
+```
+
+- You can only pass serializable data to server actions as arguments.
+- Server actions can only return serializable data.
+- Server actions run on the server and therefore are a good place to put logic that should not be exposed to the client.
+
+## CSS
+
+Expo Router supports importing global CSS and CSS modules in server components.
+
+```tsx app/index.tsx
+import './styles.css';
+import styles from './styles.module.css';
+
+export default function Index() {
+  return <div className={styles.container}>Hello, world!</div>;
+}
+```
+
+The CSS will be hoisted into the client bundle from the server.
+
+## Metadata
+
+React Server Components are a feature of React 19. To enable them, Expo CLI automatically uses a special canary build of React on all platforms. We'll remove this feature in the future when React 19 is enabled by default in React Native.
+
+As a result, you can use React 19 features such as placing `<meta>` tags anywhere in your app (web-only).
+
+```tsx app/index.tsx
+export default function Index() {
+  return (
+    <>
+      {process.env.EXPO_OS === 'web' && (
+        <>
+          <meta name="description" content="Hello, world!" />
+          <meta property="og:image" content="/og-image.png" />
+        </>
+      )}
+      <MyComponent />
+    </>
+  );
+}
+```
+
+Use this in favor of the `Head` component from `expo-router/head`.
+
+## Suspense
+
+You can stream back partial UI from the server while waiting for data to load by using React Suspense.
+
+In the following example, we'll instantly return a `Loading...` text on the client, and when the `<MediumTask>` finishes rendering one second later, it will replace the text with `Medium task done!`. The `<ExpensiveTask>` will take three seconds to load, and when it finishes, it will replace the text with `Expensive task done!`.
```

Comment:
```suggestion
In the following example, a `Loading...` text is returned instantly on the client, and when the `<MediumTask>` finishes rendering one second later, it will replace the text with `Medium task done!`. The `<ExpensiveTask>` will take three seconds to load, and when it finishes, it will replace the text with `Expensive task done!`.
```